### PR TITLE
Pulse 4125

### PR DESCRIPTION
**Pull request by martinezah**
This PR replaces adds a pluggable backend for Twitter API authentication.
```
git checkout PULSE-4125
pip install -r requirements.txt
```
## Unit Testing
To execute mocked unit tests:
```
python -m pytest tests/test_manager.py
```
If an integration environment is configured (see below), you can run online unit tests:
```
ONLINE=1 python -m pytest tests/test_manager.py
```

## Integration Testing

Ensure that you've populated `traptor/localsettings.py` with valid values for at least the following:
```
APIKEYS = {
    'CONSUMER_KEY': '___',
    'CONSUMER_SECRET': '___',
}
```
Run:
```
python traptor/manager/run.py
```
In a separate terminal, use curl to send some test requests:
```
curl -H "Content-Type: application/json" -X POST -d '{"type": "username", "value": "katyperry"}' http://localhost:5000/v1.0/validate
curl -H "Content-Type: application/json" -X POST -d '{"type": "geo", "value": "-124.84,24.39,-66.88,49.38"}' http://localhost:5000/v1.0/validate
curl -H "Content-Type: application/json" -X POST -d '{"type": "keyword", "value": "today"}' http://localhost:5000/v1.0/validate
```
Each response should contain `"status": "ok"`

### CODE REVIEW

File: traptor/settings.py [hunk 1]
```diff
@@ -51,6 +51,11 @@
 API_PORT = os.getenv('API_PORT', 5000)
 API_DIR = os.getenv('API_DIR', '.')
 API_SPEC = os.getenv('API_SPEC', 'spec.yml')
+API_BACKEND = os.getenv('API_BACKEND', 'piscina')
```
Comment (madisonb) [inline traptor/settings.py]: can we default this to 'local' for people who normally just want to use the direct twitter api?

**Comment by madisonb** [comment]
Everything worked as expected: Offline, Online, and both curl requests were OK 👍 . Code looks good, my only question is "what happens when the assert == 200 call fails?" do we have a failsafe or backup in place?

File: traptor/settings.py (hunk 1 above)
Comment (martinezah) [inline traptor/settings.py]: Fixed

**Comment by martinezah** [comment]
Error responses are abnormal but not always problematic; eg you'll get a 404 error if you lookup a username that doesn't exist. We already catch those exceptions and handle them cleanly, but I've added some detail to the `assert` so that it's easier to understand what's happening.

**Pull request merged**